some refactor on register_distributed_ops

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38657 some refactor on register_distributed_ops**
* #38494 [jit] register distributed backward
* #38656 [dist_autograd] expose distributed backward C++ API

Differential Revision: [D21940442](https://our.internmc.facebook.com/intern/diff/D21940442)

wanchaol committed:
some refactor on register_distributed_ops

[ghstack-poisoned]

wanchaol committed:
Update on "some refactor on register_distributed_ops"




[ghstack-poisoned]

wanchaol committed:
Update on "some refactor on register_distributed_ops"




[ghstack-poisoned]

wanchaol committed:
Update on "some refactor on register_distributed_ops"




[ghstack-poisoned]

wanchaol committed:
Update on "some refactor on register_distributed_ops"




[ghstack-poisoned]

wanchaol committed:
Update on "some refactor on register_distributed_ops"




[ghstack-poisoned]

xush6528 commented on torch/csrc/jit/runtime/register_ops_utils.cpp:
```diff
@@ -2,17 +2,17 @@
-}
+// c10::AliasAnalysisKind aliasAnalysisSpecialCase() {
+//   return c10::AliasAnalysisKind::INTERNAL_SPECIAL_CASE;
+// }
```
Remove them?

xush6528 commented on torch/csrc/jit/runtime/register_ops_utils.h:
```diff
@@ -34,11 +34,17 @@
-c10::AliasAnalysisKind aliasAnalysisSpecialCase();
+inline c10::AliasAnalysisKind aliasAnalysisSpecialCase() {
+  return c10::AliasAnalysisKind::INTERNAL_SPECIAL_CASE;
+}
```
What's the reason to make them inline? I though it only returns a enum element.

wanchaol replied:
it's a simple enum return so inline should make it faster in C++, also it seems like `register_ops_utils.cpp` have some issue linking the symbols with the header, so if I don't use inline, there's a linking issue when we use these function in any op registration (didn't dig into it further).

wanchaol committed:
Update on "some refactor on register_distributed_ops"




[ghstack-poisoned]

wanchaol committed:
Update on "some refactor on register_distributed_ops"




[ghstack-poisoned]

wanchaol committed:
Update on "some refactor on register_distributed_ops"




[ghstack-poisoned]

wanchaol committed:
Update on "some refactor on register_distributed_ops"




[ghstack-poisoned]

wanchaol committed:
Update on "some refactor on register_distributed_ops"


Differential Revision: [D21940442](https://our.internmc.facebook.com/intern/diff/D21940442)

[ghstack-poisoned]
